src/main/scala/interfaces/FDI.scala

```diff
@@ -0,0 +1,179 @@
+    */
+  val lpRetimerCrd = Output(Bool())
+
+  /** When asserted at a rising clock edge, it indicates a single credit return
+    * from the Retimer to the Adapter. Each credit corresponds to 256B of
+    * mainband data. This signal must NOT assert if the remote Link partner is
+    * not a Retimer.
+    */
+  val plRetimerCrd = Input(Bool())
+
+  /** Protocol layer request to Adapter to request state change. */
+  val lpStateReq = Output(ProtoStateReq())
+
+  /** Protocol Layer to Adapter indication that an error has occurred which
+    * requires the Link to go down. Adapter must propagate this request to RDI,
+    * and move the Adapter LSMs (and CXL vLSMs if applicable) to LinkError state
+    * once RDI is in LinkError state. It must stay there as long as
+    * lp_linkerror=1. The reason for having this be an indication decoupled from
+    * regular state transitions is to allow immediate action on part of the
+    * Protocol Layer and Adapter in order to provide the quickest path for error
+    * containment when applicable (for example, a viral error escalation could
+    * map to the LinkError state)
+    */
+  val lpLinkError = Output(Bool())
+
+  /** Adapter to Protocol Layer Status indication of the Interface.
+    *
+    * The status signal is permitted to transition from Adapter autonomously
+    * when applicable. For example the Adapter asserts the Retrain status when
+    * it decides to enter retraining either autonomously or when requested by
+    * remote agent.
+    */
+  val plStateStatus = Input(ProtoState())
+
+  val plInbandPres = Input(Bool())
+  val plError = Input(Bool())
+  val plCorrectableError = Input(Bool())
+  val plNonFatalError = Input(Bool())
+  val plTrainError = Input(Bool())
+
+  val plRxActiveReq = Input(Bool())
+  val lpRxActiveStatus = Output(Bool())
+
+  val plProto = Input(ProtoType())
+  val plProtoFlitFmt = Input(ProtoFlitFmt())
+  val plProtoValid = Input(Bool())
+
+  val plStallReq = Input(Bool())
+  val lpStallAck = Output(Bool())
+  val plPhyInRecenter = Input(Bool())
+  val plPhyinL1 = Input(Bool())
+  val plPhyinL2 = Input(Bool())
+  val plSpeedMode = Input(SpeedMode())
+  val plLinkWidth = Input(PhyWidth())
+
+  // Tie to 1 if clock gating not supported.
+  val plClkReq = Input(Bool())
+  val lpClkAck = Output(Bool())
+
+  // Tie to 1 if clock gating not supported.
+  val lpWakeReq = Output(Bool())
+  val plWakeAck = Input(Bool())
+
+  val plConfig = Flipped(Valid(UInt(sbWidth.W)))
+  val plConfigCredit = Input(Bool())
+  val lpConfig = Valid(UInt(sbWidth.W))
+  val lpConfigCredit = Output(Bool())
+}
+
+object ProtoStream extends ChiselEnum {
+  val stack0PCIe = Value(0x1.U(8.W))
+  val stack0CXLI = Value(0x2.U(8.W))
+  val stack0CXLC = Value(0x3.U(8.W))
+  val stack0Stream = Value(0x4.U(8.W))
+  val stack1PCIe = Value(0x11.U(8.W))
+  val stack1CXLI = Value(0x12.U(8.W))
+  val stack1CXLC = Value(0x13.U(8.W))
+  val stack1Stream = Value(0x14.U(8.W))
+}
```

Comment:
It might be worth splitting this into an enum for the protocol + a bit/field for stack number, instead of having enum values with the stack included